tests/runtests.py

```diff
@@ -465,38 +470,11 @@ def bisect_tests(bisection_label, options, test_labels, start_at, start_after):
             pass
 
     subprocess_args = get_subprocess_args(options)
-
-    iteration = 1
-    while len(test_labels) > 1:
-        midpoint = len(test_labels) // 2
-        test_labels_a = test_labels[:midpoint] + [bisection_label]
-        test_labels_b = test_labels[midpoint:] + [bisection_label]
-        print("***** Pass %da: Running the first half of the test suite" % iteration)
-        print("***** Test labels: %s" % " ".join(test_labels_a))
-        failures_a = subprocess.run(subprocess_args + test_labels_a)
-
-        print("***** Pass %db: Running the second half of the test suite" % iteration)
-        print("***** Test labels: %s" % " ".join(test_labels_b))
-        print("")
-        failures_b = subprocess.run(subprocess_args + test_labels_b)
-
-        if failures_a.returncode and not failures_b.returncode:
-            print("***** Problem found in first half. Bisecting again...")
-            iteration += 1
-            test_labels = test_labels_a[:-1]
-        elif failures_b.returncode and not failures_a.returncode:
-            print("***** Problem found in second half. Bisecting again...")
-            iteration += 1
-            test_labels = test_labels_b[:-1]
-        elif failures_a.returncode and failures_b.returncode:
-            print("***** Multiple sources of failure found")
-            break
-        else:
-            print("***** No source of failure found... try pair execution (--pair)")
-            break
-
-    if len(test_labels) == 1:
-        print("***** Source of error: %s" % test_labels[0])
+    return run_bisection(
+        bisection_label,
+        test_labels,
+        subprocess_args,
```

Comment:
We should be able to pass `pair` and `bisect` option to the `DiscoverRunner` and don't call `run_bisection()` and `run_pairing()` as an external tool.